### PR TITLE
User Settings: Ignore empty language codes & validate

### PR DESCRIPTION
**Pull request by 6543**
* [x] Ignore incorect language codes
* [x] Validate lang code

ref:
 * *https://codeberg.org/Codeberg/Community/issues/30*

### CODE REVIEW

**Comment by lafriks** [comment]
tests seems to fail now

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -246,7 +246,7 @@ social_register_helper_msg = Already have an account? Link it now!
 disable_register_prompt = Registration is disabled. Please contact your site administrator.
 disable_register_mail = Email confirmation for registration is disabled.
 remember_me = Remember this Device
-forgot_password_title= Forgot Password
+forgot_password_title = Forgot Password
```
Comment (6543) [inline options/locale/locale_en-US.ini]:
```suggestion
forgot_password_title= Forgot Password
```

File: routers/user/setting/profile.go [hunk 2]
```diff
@@ -94,7 +95,14 @@ func ProfilePost(ctx *context.Context, form auth.UpdateProfileForm) {
 	ctx.User.KeepEmailPrivate = form.KeepEmailPrivate
 	ctx.User.Website = form.Website
 	ctx.User.Location = form.Location
-	ctx.User.Language = form.Language
+	if len(form.Language) != 0 {
```
Comment (6543) [inline routers/user/setting/profile.go:98]:
```suggestion
	if len(form.Language) == 5 {
```

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
@lafriks fixed :)

**Review by mrsdizzie — APPROVED** [review]

**Pull request merged**